bin/compile

```diff
@@ -196,25 +199,41 @@ fi
 export YARN_CACHE_FOLDER NPM_CONFIG_CACHE
 
 install_bins() {
-  local node_engine npm_engine yarn_engine npm_version node_version
+  local node_engine npm_engine yarn_engine node_version package_manager
 
   node_engine=$(read_json "$BUILD_DIR/package.json" ".engines.node")
   npm_engine=$(read_json "$BUILD_DIR/package.json" ".engines.npm")
   yarn_engine=$(read_json "$BUILD_DIR/package.json" ".engines.yarn")
+  package_manager=$(read_json "$BUILD_DIR/package.json" ".packageManager")
 
   meta_set "node-version-request" "$node_engine"
   meta_set "npm-version-request" "$npm_engine"
   meta_set "yarn-version-request" "$yarn_engine"
+  meta_set "package-manager-request" "$package_manager"
+
+  echo "engines.node (package.json):   ${node_engine:-unspecified}"
+  echo "engines.npm (package.json):    ${npm_engine:-unspecified (use default)}"
 
-  echo "engines.node (package.json):  ${node_engine:-unspecified}"
-  echo "engines.npm (package.json):   ${npm_engine:-unspecified (use default)}"
   if $YARN || [ -n "$yarn_engine" ]; then
-    echo "engines.yarn (package.json):  ${yarn_engine:-unspecified (use default)}"
+    echo "engines.yarn (package.json):   ${yarn_engine:-unspecified (use default)}"
   fi
+
+  if [ -n "$package_manager" ]; then
+    echo "packageManager (package.json): $package_manager"
+  fi
+
   echo ""
 
   warn_node_engine "$node_engine"
 
+  if [ -n "$yarn_engine" ] && [[ "$package_manager" == yarn* ]]; then
+    warn "The Yarn version is specified in both the 'packageManager' and 'engines.yarn' field in package.json. The 'packageManager' field will be preferred and $package_manager will be used."
+  fi
+
+  if has_release_script "$BUILD_DIR"; then
+    warn "A yarn release script is configured using 'yarnPath' in .yarnrc.yml. This will be used instead of the version of Yarn specified in the 'packageManager' field in package.json."
```

Comment:
Is `warn` the correct function here? This messaging doesn't sound like a warning, more like information.

Reply:
Wouldn't this message also apply when folks had specified `yarnPath` and `engines.yarn`? I think we ignore `engines.yarn` in that scenario too?

Reply:
Similar to the comment above, I think `warn` is appropriate here. In the presence of conflicting signals I want to make sure it's clear to the user why their build is behaving a certain way and how they eliminate this ambiguity.  In this scenario, Yarn uses the `packageManager` version installed to bootstrap the version from the configured `yarnPath` which may not be the desired behavior. 

So, it won't fail the build but if the user expects Yarn `x.y.z` and instead got Yarn `a.b.c` it should be clear from the build logs why that may have occurred and how to fix it.

I notice that I don't have a well defined "action" for the user in this message though so I'll add that.

Reply:
@joshwlewis I've changed this warning to provide more context and clearer actions.